src/parser.rs

```diff
@@ -2624,6 +2626,8 @@ impl<'a> Parser<'a> {
                 return_type,
                 params,
             })
+        } else if dialect_of!(self is DuckDbDialect) {
```

Comment:
Can you please also allow `GenericDialect` as well (which typically allows the superset of everything supported by sqlparser)?

Reply:
I allowed GenericDialect for `CREATE MACRO` but not for their `CREATE FUNCTION` alias.
Mostly because duckdb's macros don't seem like a common denominator for SQL functions (if there is any).
More specifically because SQL functions are typically typed while duckdb macros are not.

I could adjust the failing test https://github.com/MartinNowak/sqlparser-rs/blob/088b2fe0759311801740d5ba490d7acb376a4e4a/tests/sqlparser_hive.rs#L268-L278 and try to merge with the existing FunctionArg (using an optional type), but presumably the AST is going the end up very messy and code will need plenty of dialect switches/config options.